pdfminer/pdfpage.py

```diff
@@ -120,15 +121,18 @@ def search(obj, parent):
     @classmethod
     def get_pages(cls, fp,
                   pagenos=None, maxpages=0, password='',
-                  caching=True, check_extractable=True):
+                  caching=True):
         # Create a PDF parser object associated with the file object.
         parser = PDFParser(fp)
         # Create a PDF document object that stores the document structure.
         doc = PDFDocument(parser, password=password, caching=caching)
-        # Check if the document allows text extraction. If not, abort.
-        if check_extractable and not doc.is_extractable:
-            error_msg = 'Text extraction is not allowed: %r' % fp
-            raise PDFTextExtractionNotAllowed(error_msg)
+        # Check if the document allows text extraction.
+        # If not, warn the user and proceed.
+        if not doc.is_extractable:
+            warning_msg = 'This PDF contains a metadata field indicating'\
+                          'that it should not allow text extraction.'    \
+                          'Ignoring this field and proceeding.'
+            warnings.warn(warning_msg, PDFTextExtractionNotAllowedWarning)
```

Comment:
Raise error if `check_extractable` and raise warning if not `check_extractable`

Reply:
Done